KIPs/kip-2.md

```diff
@@ -0,0 +1,252 @@
+
+The specification of Klaytn keystore format v4 in JSON for a Klaytn account is shown below.
+This format can express various Klaytn account key types such as a single key, multiple keys, and role-based keys. 
+
+```
+<keystore> :=
+{
+  "version": 4,
+  "id": <a 128-bit UUID in hexstring>,
+  "address": <Klaytn address, a 160-bit hexstring>,
+  "keyring": <keyring>
+}
+
+<keyring> := [<key> (, <key>)*] | [<keys> (, <keys>)*]
+
+<keys> := [] | [<key> (, <key>)*]
+
+<key> :=
+{
+  "cipher": "aes-128-ctr",
+  "ciphertext": <a 256-bit hexstring>,
+  "cipherparams": {
+    "iv": <a 128-bit hexstring>
+  },
+  "kdf": "scrypt" | "pbkdf2",
+  "kdfparams": <kdfparams-scrypt> | <kdfparams-pbkdf2>,
+  "mac": <a 256-bit hexstring>
+}
+
+<kdfparams-scrypt> :=
+{
+  "dklen": <an integer>,
+  "salt": <a 256-bit hexstring>,
+  "n": <an integer>,
+  "r": <an integer>,
+  "p": <an integer>,
+}
+
+<kdfparams-pbkdf2> :=
+{
+  "dklen": <an integer>,
+  "salt": <a 256-bit hexstring>,
+  "prf": "hmac-sha256",
+  "c": <an integer>,
+}
+
+```
+
+## Examples
+Examples are as following:
+
+1. A single key
+    ```
+    {
+        "version": 4,
+        "id": "b6c6bd8f-e3ba-43c1-af4b-1b7dfe13e5f9",
+        "address": "0xa8677855999712d23692a4cad77bb9396a1174cd",
+        "keyring": [
+            {
+                "ciphertext": "5cbd3cf235d02718f1cfac84157786d8e36fcf651e15f0c9a4303a2338697126",
+                "cipherparams": {
+                    "iv": "92a89f3011e020f6a70bb34fddb006c5"
+                },
+                "cipher": "aes-128-ctr",
+                "kdf": "scrypt",
+                "kdfparams": {
+                    "dklen": 32,
+                    "salt": "aa38c864f4f6f538dc3bb4961c9d5d5772b163d2243aa8472cc4f83338e67cd3",
+                    "n": 4096,
+                    "r": 8,
+                    "p": 1
+                },
+                "mac": "3355abc00f3fb1ae06821ece9c13359a0da1bd284cf29ffc26a534a40d210f28"
+            }
+        ]
+    }
+    ```
+2. Multiple keys
+    ```
+    {
+        "version": 4,
+        "id": "b6c6bd8f-e3ba-43c1-af4b-1b7dfe13e5f9",
+        "address": "0xa8677855999712d23692a4cad77bb9396a1174cd",
+        "keyring": [
+            {
+               "ciphertext": "5cbd3cf235d02718f1cfac84157786d8e36fcf651e15f0c9a4303a2338697126",
+               "cipherparams": {
+                   "iv": "92a89f3011e020f6a70bb34fddb006c5"
+               },
+               "cipher": "aes-128-ctr",
+               "kdf": "scrypt",
+               "kdfparams": {
+                   "dklen": 32,
+                   "salt": "aa38c864f4f6f538dc3bb4961c9d5d5772b163d2243aa8472cc4f83338e67cd3",
+                   "n": 4096,
+                    "r": 8,
+                    "p": 1
+                 },
+                "mac": "3355abc00f3fb1ae06821ece9c13359a0da1bd284cf29ffc26a534a40d210f28"
+            },
+            {
+               ciphertext: '5e2f95f61d7af3bebf4ff9f5d5813690c80b0b5aaebd6e8b22d0f928ff06776a',
```

Comment:
When showing an example, it is better to unify the form of the example, either in the form of a Json string or in the form of an object parsed by Json.

Reply:
I also added to quotes to all the keys. Please take a look. :)